src/sync/installation.ts

```diff
@@ -240,38 +261,37 @@ export const processInstallation =
 
 			const processor = tasks[task];
 
-			const handleGitHubError = (err) => {
-				if (err.errors) {
-					const ignoredErrorTypes = ["MAX_NODE_LIMIT_EXCEEDED"];
-					const notIgnoredError = err.errors.filter(
-						(error) => !ignoredErrorTypes.includes(error.type)
-					).length;
-
-					if (notIgnoredError) {
-						throw err;
-					}
-				} else {
-					throw err;
-				}
-			};
-
 			const execute = async () => {
 				if (await booleanFlag(BooleanFlags.SIMPLER_PROCESSOR, true)) {
-					try {
-						return await processor(github, repository, cursor, 20);
-					} catch (err) {
-						logger.error({ err, job, github, repository, cursor, task }, "Error Executing Task");
-						handleGitHubError(err);
-					}
+
+					// just try with one page size
+					return await processor(github, repository, cursor, 20);
+
 				} else {
+
 					for (const perPage of [20, 10, 5, 1]) {
+						// try for decreasing page sizes in case GitHub returns errors that should be retryable with smaller requests
 						try {
 							return await processor(github, repository, cursor, perPage);
 						} catch (err) {
-							logger.error({ err, job, github, repository, cursor, task }, "Error Executing Task");
-							handleGitHubError(err);
+							logger.error({
+								err,
+								job,
+								github,
+								repository,
+								cursor,
+								task
+							}, `Error processing job with page size ${perPage}, retrying with next smallest page size`);
+							if (isRetryableWithSmallerRequest(err)) {
+								// error is retryable, retrying with next smaller page size
+								continue;
+							} else {
+								// error is not retryable, re-throwing it
+								throw err;
+							}
```

Comment:
This is the main change. Made the logic around the error handling a lot easier to understand, hopefully. This code is covered with the `SIMPLER_PROCESSOR` flag, so this change will only take effect, once I change the feature flag back to `false`.